src/xpk/core/blueprint/definition.py

```diff
@@ -0,0 +1,59 @@
+"""
```

Comment:
nit: maybe plural name? `definitions.py`

nit: what do you think if we name them `blueprint_definitions.py` and `blueprint_generator.py`? the class name it BlueprintGenerator, right? I don't have a strong preference here, so ignore it if you are see current naming better.